src/main/java/io/cdap/plugin/condition/Conditional.java

```diff
@@ -187,7 +183,18 @@ public ConditionConfig(String expression) {
     String getExpression() {
       return expression;
     }
-  }
 
+    public void validate(EL el, FailureCollector collector) {
+      if (!containsMacro("expression") && !Strings.isNullOrEmpty(expression)) {
+        try {
+          el.compile(expression);
+        } catch (ELException e) {
+          collector.addFailure(String.format("Error encountered while compiling the expression : %s", e.getMessage()),
+                               "Ensure a valid expression is provided.")
```

Comment:
don't think we need a corrective action. Especially if the exception message is large, this will look weird.